.github/workflows/build-notebooks-pr.yaml

```diff
@@ -13,8 +16,16 @@ jobs:
       matrix: ${{ steps.gen.outputs.matrix }}
     steps:
       - uses: actions/checkout@v4
-      - run: python3 ci/cached-builds/gen_gha_matrix_jobs.py
+
+      - run: |
+          python3 ci/cached-builds/gen_gha_matrix_jobs.py \
```

Comment:
Maybe get a list of changed files using `git diff` and pass them to the script?

Reply:
`- uses: actions/checkout@v4` by default does a shallow clone; to do `git diff`, I need to either clone entire git history (that's 70MB+ here) or I need to clone last 20 and hope nobody puts more than 20 commits into a PR (but I would detect this and fail the ci in this case) or I can somehow determine the depth first...

asking gh api seemed easier to me to begin with; I'm not opposed to `git diff`, just that I don't think it is better than what I have now

Reply:
Not sure we need all that but I tend to minimize the number of external calls in build processes in favor of things that can be retrieved internally to avoid issues. Let's keep an eye on it. If it becomes an issue, we can use something like the code below:

```yaml
    steps:
      - name: Checkout
        uses: actions/checkout@v4

      - name: Diff
        shell: bash
        run: |
          git fetch origin ${{ github.base_ref }}

          base_sha=$(git rev-parse origin/${{ github.base_ref }})
          head_sha=$(git rev-parse ${{ github.sha }})

          changed_files=($(git diff --name-only $base_sha $head_sha))

          echo "Changed files:"
          for file in "${changed_files[@]}"; do
              echo "$file"
          done
```

Reply:
```
git fetch origin ${{ github.base_ref }}
```

Right, this is nice! I'll try to keep this in mind; at the least it is portable between git hosting services, which gh api sure is not.

Reply:
Logged a ticket for it in GitHub, will do that later

* https://github.com/opendatahub-io/notebooks/issues/560